packages/api/src/command/medical/document/document-bulk-download.ts

```diff
@@ -0,0 +1,170 @@
+import { DocumentBulkDownloadDTO } from "../../../routes/medical/dtos/document-bulk-downloadDTO";
+
+import {
+  DocumentBulkDownloadProgress,
+  isDocBulkDownloadProcessing,
+  DocumentDownloadStatus,
+} from "../../../domain/medical/document-bulk-download";
+//import { getSignedUrls } from "@metriport/core/external/aws/lambda-logic/document-bulk-signing";
+import { Util } from "../../../shared/util";
+import { getPatientOrFail } from "../patient/get-patient";
+import { Patient } from "../../../domain/medical/patient";
+import { uuidv7 } from "@metriport/core/util/uuid-v7";
+import { storeBulkDownloadQueryInit } from "../patient/query-init";
+import { appendDocBulkDownloadProgress } from "../patient/append-bulk-doc-download-progress";
+import { DocumentReference } from "@medplum/fhirtypes";
+import { makeLambdaClient } from "../../../external/aws/lambda";
+import { getLambdaResultPayload } from "@metriport/core/external/aws/lambda";
+import {
+  DocumentBulkSignerLambdaRequest,
+  DocumentBulkSignerLambdaResponse,
+} from "@metriport/core/external/aws/lambda-logic/document-bulk-signing";
+
+const BATCH_SIZE = 100;
+const lambdaClient = makeLambdaClient();
+const bulkSigningLambdaName = Config.getBulkUrlSigningLambdaName();
+
+/**
+ * The function `triggerBulkUrlSigning` triggers the bulk signing process lambda for a patient's documents and
+ * returns the progress of the bulk download. It also triggers the webhook for the bulk download.
+ * @param {string} cxId - cxId
+ * @param {string} patientId - patientId
+ * @returns a Promise that resolves to a DocumentBulkDownloadProgress object.
+ */
+export const triggerBulkUrlSigning = async (
+  cxId: string,
+  patientId: string
+): Promise<DocumentBulkDownloadProgress> => {
+  if (!bulkSigningLambdaName) throw new Error("Bulk Signing Lambda Name is undefined");
+  const { log } = Util.out(`triggerBulkSigning - M patient ${patientId}`);
+
+  const patient = await getPatientOrFail({ id: patientId, cxId });
+
+  const docBulkDownloadProgress = patient.data.documentBulkDownloadProgress;
+  const requestId = getOrGenerateRequestId(docBulkDownloadProgress);
+
+  if (isDocBulkDownloadProcessing(docBulkDownloadProgress?.download)) {
+    log(
+      `Patient ${patientId}, Request ${requestId}, docBulkDownloadProgress is already 'processing', skipping...`
+    );
+    return createBulkDownloadQueryResponse("processing", patient);
+  }
+
+  const documents = await searchDocuments({ cxId, patientId });
+
+  let updatedPatient = await storeBulkDownloadQueryInit({
+    id: patient.id,
+    cxId: patient.cxId,
+    documentBulkDownloadProgress: { download: { status: "processing" } },
+    requestId,
+    totalDocuments: documents.length,
+  });
+
+  let successes = 0;
+  const errors = 0;
+  // Process each batch
+  const batches = chunk(documents, BATCH_SIZE);
+  for (let batchIndex = 0; batchIndex < batches.length; batchIndex++) {
+    const batch = batches[batchIndex] as DocumentReference[];
```

Comment:
NABD, but could use `Object.entries(batches)`, it returns a typed result and we don't need to use `as`.